### PR TITLE
Do not do any analytics requests if analytics are disabled

### PR DESCRIPTION
**Pull request by akx**
# Description

* relevant motivation
  * https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/10324
  * https://github.com/oobabooga/text-generation-webui/blob/c746a5bd00c39776bec0091a570f3348aa785caa/server.py#L12-L16
* a summary of the change 
  * Gradio should not call home if the user asks to disable analytics. That should be pretty self-evident.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added a short summary of my change to the CHANGELOG.md
- [x] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by abidlabs** [comment]
Thanks for creating the PR @akx. I don't follow the original issue. In the `Blocks` class, we already check if `self.analytics_enabled` before making any call to the analytics methods: `utils.error_analytics()`, `utils.launched_telemetry()`, `utils.initiated_analytics(), etc.. In other words, this PR would not have any effect. Am I missing something?

**Comment by akx** [comment]
Please see the linked issues. People are seeing Gradio do network calls at initial import and working around that.

**Comment by akx** [comment]
@abidlabs To elaborate:

* In particular, `Interface.__init__()` seems to unconditionally call `version_check()`, which calls home to `https://api.gradio.app/pkg-version`.
* All of the analytics functions also use Amazon's https://checkip.amazonaws.com/ service, another third party. (Furthermore, I don't think it's quite OK that supposedly anonymous analytics data also includes a public IP address.)

**Comment by akx** [comment]
Done, @abidlabs

**Comment by abidlabs** [comment]
Thanks for the important fix @akx and the cleanup. LGTM 

I'd like to get a review by @dawoodkhan82 and then we should be good to merge in.

**Review by abidlabs — APPROVED** [review]

File: gradio/analytics.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    threading.Thread(
+        target=_do_analytics_request,
+        kwargs={
+            "url": f"{ANALYTICS_URL}gradio-initiated-analytics/",
+            "data": {**data, "ip_address": get_local_ip_address()},
+        },
+    ).start()
+
+
+def launched_telemetry(blocks: gradio.Blocks, data: dict[str, Any]) -> None:
```
Comment (dawoodkhan82) [inline gradio/analytics.py:97]: Can we rename this to `launched_analytics`

File: gradio/interface.py [hunk 2]
```diff
@@ -362,7 +362,7 @@ def __init__(
         self.local_url = None
 
         self.favicon_path = None
-        utils.version_check()
+        analytics.version_check()
```
Comment (dawoodkhan82) [inline gradio/interface.py:365]: This check won't work now if analytics is disabled. @abidlabs

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,8 @@
 
 ## Bug Fixes:
 
+- Fix "TypeError: issubclass() arg 1 must be a class" When use Optional[Types] by [@lingfengchencn](https://github.com/lingfengchencn) in [PR 4200](https://github.com/gradio-app/gradio/pull/4200). 
+- Gradio will no longer send any analytics if analytics are disabled with the GRADIO_ANALYTICS_ENABLED environment variable. By [@akx](https://github.com/akx) in [PR 4194](https://github.com/gradio-app/gradio/pull/4194)
```
Comment (dawoodkhan82) [inline CHANGELOG.md:12]: Were analytics being sent when the env variable was set to false before this PR?

**Review by dawoodkhan82 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (abidlabs) [reply CHANGELOG.md:12]: What we call analytics weren't being sent, but network requests were being made to the Gradio server, see https://github.com/gradio-app/gradio/pull/4194#issuecomment-1546937585

File: gradio/interface.py (hunk 2 above)
Comment (abidlabs) [reply gradio/interface.py:365]: Yes I think that's an acceptable sacrifice to ensure no network requests to gradio servers

File: gradio/analytics.py (hunk 1 above)
Comment (abidlabs) [reply gradio/analytics.py:97]: Agreed, I'll make the change

**Comment by abidlabs** [comment]
Thanks @dawoodkhan82 for giving this a look. Renamed the function and we should be good for a merge once CI passes

**Pull request merged**